node-tests/acceptance/build-test.js

```diff
@@ -398,5 +398,61 @@ describe('Acceptance', function() {
       output.doesNotContain(`engines-dist/${engineName}/assets/engine-vendor.js`, moduleMatcher(`${addonName}/templates/components/foo`));
       output.doesNotContain(`engines-dist/${engineName}/assets/engine-vendor.css`, cssCommentMatcher(`${addonName}.css`));
     }));
+
+    it('duplicate addons in lazy engines when tree is not the same', co.wrap(function* () {
```

Comment:
This would be clearer as: `duplicates an addon in lazy engines when the tree returned by the addon is different`